backend/src/controllers/notes.ts

```diff
@@ -11,17 +11,30 @@ export const getNotes: RequestHandler = async (req, res, next) => {
   try {
     assertIsDefined(authenticatedUserId);
 
-    const notes = await NoteModel.find({ userId: authenticatedUserId }).exec();
+    const notes = await NoteModel.find({
+      userId: authenticatedUserId,
+      deletedAt: null,
+    }).exec();
     res.status(200).json(notes);
   } catch (error) {
     next(error);
   }
 };
 
+const GetNoteParamsSchema = z.object({
```

Comment:
We should have a separate file for the schemas, they don't belong the controller.